source/common/stats/thread_local_store.h

```diff
@@ -270,6 +277,14 @@ class ThreadLocalStoreImpl : Logger::Loggable<Logger::Id::stats>, public StoreRo
     ThreadLocalStoreImpl& parent_;
     const std::string prefix_;
     CentralCacheEntry central_cache_;
+
+    // It's ok to cache information relating to the top-level stat name filter regex, since it is
+    // set once at startup and never modified.
+    const bool has_filter_;
```

Comment:
I think there's redundant info being stored here. I would either store a `bool` and `std::regex` or store `absl::optional<std::regex>` and use the .has_value() where `has_filter_` is used now.

Reply:
I will replace this with the stored optional regex `filter_` and a call to `filter_.has_value()`.